diff --git a/node_modules/expo-keep-awake/src/index.ts b/node_modules/expo-keep-awake/src/index.ts
index b7a856e..82385ba 100644
--- a/node_modules/expo-keep-awake/src/index.ts
+++ b/node_modules/expo-keep-awake/src/index.ts
@@ -30,11 +30,20 @@ export function useKeepAwake(tag?: string, options?: KeepAwakeOptions): void {
 
   useEffect(() => {
     let isMounted = true;
-    activateKeepAwakeAsync(tagOrDefault).then(() => {
-      if (isMounted && ExpoKeepAwake.addListenerForTag && options?.listener) {
-        addListener(tagOrDefault, options.listener);
-      }
-    });
+    const activation = activateKeepAwakeAsync(tagOrDefault);
+
+    activation
+      .then(() => {
+        if (isMounted && ExpoKeepAwake.addListenerForTag && options?.listener) {
+          addListener(tagOrDefault, options.listener);
+        }
+      })
+      .catch((error) => {
+        if (!options?.suppressDeactivateWarnings) {
+          throw error;
+        }
+      });
+
     return () => {
       isMounted = false;
       if (options?.suppressDeactivateWarnings) {
